keeper-gradle-plugin/src/main/java/com/slack/keeper/ZipFlingerExt.kt

```diff
@@ -30,7 +30,12 @@ internal fun File.classesSequence(): Sequence<Pair<String, File>> {
     .filter { it.extension == "class" }
     .filterNot { "META-INF" in it.name }
     .sortedBy { it.invariantSeparatorsPath }
-    .map { it.absolutePath.removePrefix(prefix).removePrefix("/") to it }
+    .map {
+      it.absolutePath
+        .removePrefix(prefix)
+        .removePrefix(File.separator)
+        .replace(File.separator, "/") to it
```

Comment:
I asked about this on the other PR too, but this seems counterintuitive to me. Isn't the "/" character the problematic one? If so, why are we adding it back manually here? Or is it because this is a zip and we need to force it to always use "/" regardless of the source OS?

Reply:
It's the latter I believe - because it's a zip

Though, I haven't tested it, I could play around with different versions of it to see conclusively but not today

The testing around this is ace now though - that was a bear!

Reply:
oh wait, found it. zip specification

https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT

item 4.4.17.1

```

   4.4.17 file name: (Variable)

       4.4.17.1 The name of the file, with optional relative path.
       The path stored MUST NOT contain a drive or
       device letter, or a leading slash.  All slashes
       MUST be forward slashes '/' as opposed to
       backwards slashes '\' for compatibility with Amiga
       and UNIX file systems etc.  If input came from standard
       input, there is no file name field.  
```

Reply:
@ZacSweers any code that causes persistent doubts yet is based on a specification deserves a good comment, so this morning as I got moving again I took that specification chunk and sprinkled it as comments with specification reference in that code. Future maintainers will hopefully rejoice